src/components/DataCollector/createConfigValidators.js

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import {
+  required,
+  validDomain,
+  allValidate,
+  createUnique
+} from "../../utils/configValidators";
+
+export default () => {
+  return {
+    propertyId: {
+      validate: allValidate(required, createUnique())
+    },
+    edgeDomain: {
+      validate: validDomain,
+      defaultValue: "alpha.konductor.adobedc.net"
+    },
+    imsOrgId: {
+      validate: allValidate(required, createUnique())
```

Comment:
I'd rather see this type of format: `validate: string().required().unique()`, but there's a JIRA issue to make validators more composable that could cover that work if we choose to follow that format: https://jira.corp.adobe.com/browse/CORE-32266. One of the advantages of that format is that it naturally scopes certain validation functionality to its respective value type. In other words, as a developer I can do this:

`validate: number().minimum(500)`

but not this:

`validate: string().minimum(50)`

`minimum` doesn't really apply to a string value, but it does a number value. If one of us tried to do `string().minimum(50)`, we'd get an error saying that `minimum` is undefined, which is what I would want. Whereas, if we did `allValidate(minimum(50), string)` our `minimum` validator would require type checking (or worse, implicitly cast the value to a number), which should be the `string` validator's job instead.

Reply:
@Aaronius / @jonsnyder, do you want to implement this change in this PR and get it over with?

Reply:
The amount of refactoring does seem to be mounting. I'd like to see it done sooner rather than later, whether @jonsnyder does it or I do it.

Reply:
I'll take a stab at it now.

Reply:
There is an old commit in the validation logic that accepts an array of validators. We could resurrect that one to support what allValidate does.